specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,353 @@
+  /// valid origins.
+  /// getTextureStream() will fail unless the requesting frame's origin URL is
+  /// added to the request filter.
+  HRESULT AddRequestedFilter([in] LPCWSTR origin);
+  /// Remove added origin, which was added by AddRequestedFilter.
+  HRESULT RemoveRequestedFilter([in] LPCWSTR origin);
+  /// Listens for stream requests from the Javascript's getTextureStream call
+  /// for the given stream id. It is called for the first request only, the
+  /// subsequent requests of same stream id will not be called.
+  /// It is expected that the host provides the stream within 10s after
+  /// being requested. The first call to Present() fulfills the stream request.
+  HRESULT add_StartRequested(
+      [in] ICoreWebView2StagingTextureStreamStartRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_StartRequested(
+      [in] EventRegistrationToken token);
+  /// Listen to stop stream request once the stream started.
+  /// It is called when user stop all streaming requests from
+  /// the renderers (Javascript) or the host calls the Stop API. The renderer
+  /// can stream again by calling the streaming request API.
+  /// The renderer cleared all registered buffers before sending
+  /// the stop request event so that the callback of the next start request
+  /// should register the textures again.
+  /// The event is triggered when all requests for given stream id closed
+  /// by the Javascript, or the host's Stop API call.
+  HRESULT add_StopRequested(
+      [in] ICoreWebView2StagingTextureStreamStopRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for stop stream request.
+  HRESULT remove_StopRequested(
+      [in] EventRegistrationToken token);
+  /// Creates shared buffer that will be referenced by the host and the browser.
+  /// By using the shared buffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal buffer id, which is the identity of the shared buffer.
+  /// The shared buffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new shared buffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+  HRESULT CreateBuffer(
+    [in] UINT32 width,
+    [in] UINT32 height,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// GetAvailableBuffer can be called on any thread like SetBuffer.
```

Comment:
Great note that this can be called on any thread. Please make sure all methods and properties that work on other threads are noted as working on other threads. Its stated generally that WebView2 methods only work on the WebView2 UI thread so we need to explicitly call out anything that doesn't work like that.